torch/_inductor/codegen/cpp_wrapper_cuda.py

```diff
@@ -264,6 +264,11 @@ def generate_kernel_call(
 
         self.generate_load_kernel_once(name, mangled_name, cubin_path, shared_mem)
 
+        # args with value 1 are added into equal_to_1 and constants
+        # in triton_meta (in the Python codegen) which makes them
+        # inlined in the PTX and compiled CUBIN
+        call_args = [arg for arg in call_args if arg != 1]
```

Comment:
not a blocker, but just a random thought - instead of skipping ```arg == 1``` cases, would it possible to not generate those args whose value equals ```1```? My worry is that it could potentially introduce mismatch (e.g.) if the same call_args is used somewhere else but we didn't filter out ```arg == 1``` arguments.

Second, I assume that all ```arg == 1``` will be added into ```equal_to_1``` meta? Thanks.

Reply:
Agreed it would be better to use the hints to guarantee there isn't a mismatch.

Reply:
> Second, I assume that all arg == 1 will be added into equal_to_1 meta?

This is correct. That's why we are excluding them here with certainty.

> would it possible to not generate those args whose value equals 1? 

I don't think we can skip those args in the upstream generation (before the codegen), as the 1-args must still be passed in the Python wrapper, but skipped in the C++ wrapper (due to the PTX nuance mentioned above). Could you point to other places in the C++ codegen where we might still need to exclude 1-args? Thanks!

Reply:
> > Second, I assume that all arg == 1 will be added into equal_to_1 meta?
> 
> This is correct. That's why we are excluding them here with certainty.
> 
> > would it possible to not generate those args whose value equals 1?
> 
> I don't think we can skip those args in the upstream generation (before the codegen), as the 1-args must still be passed in the Python wrapper, but skipped in the C++ wrapper (due to the PTX nuance mentioned above). Could you point to other places in the C++ codegen where we might still need to exclude 1-args? Thanks!

Thanks for the clarification. I am not aware of any other use of ```call_args``` in the current cuda wrapper code. It's just my gut-feeling that dropping args based on unguarded assumptions seems to be dangerous :)